lib/game.rb

```diff
@@ -0,0 +1,101 @@
+require_relative './Board'
+# This is a class which all of the methods about user interface will be in
+class Game
+  attr_accessor :winner, :board
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @board = Board.new(@player1, @player2)
+    @winner = ''
+  end
+
+  def input_is_valid?(user_move)
+    if user_move.to_i >= 1 && user_move.to_i <= 9
+      if @board.blank?(user_move)
+        true
+      else
+        puts 'That location is full, please enter the blank location!'
+        false
+      end
+    else
+      false
+    end
+  end
+
+  def any_empty?
+    @board.board.values.any?(' ')
+  end
+
+  def move(player)
+    puts "It is #{player.name}'s turn to play now."
+    puts "#{player.name}, please enter location(1-9) to puts x there!"
```

Comment:
If we move these prints to the main file, we must write that code 2 times, so it will not obey with the DRY rule which is also part of the requirements in the project. (DRY = Don't Repeat Yourself)

Reply:
You don't have to write the code twice. Instead of using `puts` you can return a string and use it in `main.rb`